File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # Input text from user
    text = get_string("Text: ")

    # Caluculate the number of letters per 100 words
    L = (count_letters(text) / count_words(text)) * 100
    # Caluculate the number of sentences per 100 words
    S = (count_sentences(text) / count_words(text)) * 100

    # Caluculate the Coleman-Liau index
    index = round(.0588 * L - 0.296 * S - 15.8)

    if index >= 16:
        print("Grade 16+")

    elif index < 1:
        print("Before Grade 1")

    else:
        print("Grade {}" .format(index))


# Count the number of letters
def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if str.isalpha(text[i]):
            letters += 1

    return letters


# Count the number of words
def count_words(text):
    words = 1
    for i in range(len(text)):
        if text[i] == " ":
            words += 1

    return words


# Count the number of sentences
def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    string text = get_string("Text: ");

    double L = (count_letters(text) / (double) count_words(text)) * 100;
    double S = (count_sentences(text) / (double) count_words(text)) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

    return 0;
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    return sentences;
}